src/closed.jl

```diff
@@ -0,0 +1,54 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
```

Comment:
I'm not super worried about the conflict with ClosedIntervals.jl since it isn't METADATA-registered (and would take some work to get ready for registration) nor does it appear to be all that actively developed.